### PR TITLE
Added state and process field

### PR DESCRIPTION
**Pull request by mkk19**
Solves https://github.com/Financial-Times/splunk-heroku/issues/31

### CODE REVIEW

**Comment by sjparkinson** [comment]
Just running this in regex101 and testing these out, my understanding is we'll be capturing the following as fields:

image

So for the `State` field, the value will always be `State` as that is the regex capture group! What you'll need to consider is where is the capture group in the regex, currently it captures the string "State" but ideally it'll capture the right hand side of the message i.e. "up" or "complete".

I've added some tests to https://regex101.com/r/vMKHxn/1, that might help know when you've got the right regex. And shout if you want to pair through this, but there's no rush either if you want to pick away at this solo too!

**Comment by mkk19** [comment]
> Just running this in regex101 and testing these out, my understanding is we'll be capturing the following as fields:
> 
> image
> 
> So for the `State` field, the value will always be `State` as that is the regex capture group! What you'll need to consider is where is the capture group in the regex, currently it captures the string "State" but ideally it'll capture the right hand side of the message i.e. "up" or "complete".
> 
> I've added some tests to https://regex101.com/r/vMKHxn/1, that might help know when you've got the right regex. And shout if you want to pair through this, but there's no rush either if you want to pick away at this solo too!

Hi Sam,

Thanks a lot for the clarification,  I've made the changes in latest commit and hopefully that is fine!

File: app/default/props.conf [hunk 1]
```diff
@@ -25,6 +25,8 @@ description = Messages about actions taken by the Heroku platform.
 
 EXTRACT-heroku_component,dyno = ^\S+\s(?<component>\S+)\s(?<dyno>\S+)
 EXTRACT-message = \s-\s(?<message>.*)
+EXTRACT-state = State changed from \w+ to (?<State>\w+)$
+EXTRACT-exit_code = Process exited with status (?<Exit_Code>\w+)$
```
Comment (sjparkinson) [inline app/default/props.conf:29]: Nailed it!

I'd lowercase the field names to be consistent with our other field names, but this looks brillant!

```suggestion
EXTRACT-state = State changed from \w+ to (?<state>\w+)$
EXTRACT-exit_code = Process exited with status (?<exit_code>\w+)$
```

**Review by sjparkinson — APPROVED** [review]
One suggestion to be consistent, looks great overall, nice one!

Let me know if you want a hand deploying it, but feel free to have a go at it too, this app doesn't restart Splunk or cause anything impactful.

**Pull request merged**